### PR TITLE
Adds a Git FAQ page

### PR DESCRIPTION
**Pull request by Benuuu**

### CODE REVIEW

**Review by paulrolfe — APPROVED** [review]

File: git/versioning-faq.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+--------------------------------------------------------------------------------
+
+### Workflow
+
+For further information on this see [this sherpa page.](https://github.com/
+IntrepidPursuits/sherpa/blob/master/git/git_workflow_quick_reference.md)
+
+#### When should I branch?
+Generally speaking, when you need to work in parallel with another ongoing set
+of work.
```
Comment (bobgilmore) [inline git/versioning-faq.md:125]: I feel that this is a little *too* lenient. The correct answer is *always*. 

But in all seriousness, this implies that one person, working solo, can work on `master` or `develop` or whatever. But even *then* there should be a distinct branch, for PR review purposes if nothing else.

**Review by bobgilmore — APPROVED** [review]
One teeny little comment about branch creation, but overall 👍

File: git/versioning-faq.md (hunk 1 above)
Comment (Benuuu) [reply git/versioning-faq.md:125]: Not sure I still have write access here.

@bobgilmore how about instead just saying:
```
#### When should I branch?
Almost always.
... <list as is>
```

Comment (bobgilmore) [reply git/versioning-faq.md:125]: Looks good to me, for what that's worth :-) Not certain that I have write access, either. Maybe I'll try to modify the branch tomorrow and see what happens ;-)

**Comment by bobgilmore** [comment]
Don’t worry about my comment, not really important.

**Pull request merged**